src/shared/tests.c

```diff
@@ -42,5 +42,9 @@ const char* get_exe_relative_testdata_dir(void) {
 
         assert_se(readlink_and_make_absolute("/proc/self/exe", &exedir) >= 0);
         assert_se(snprintf(testdir, sizeof(testdir), "%s/testdata", dirname(exedir)) > 0);
+        if (access(testdir, F_OK) < 0) {
+                fprintf(stderr,  "Test data directory '%s' does not exist, set $TEST_DIR\n", testdir);
+                exit(1);
```

Comment:
`exit(1)` prevents all cleanup actions.

For example, `test_engine` creates the directory:
```c

        assert_se(runtime_dir = setup_fake_runtime_dir());

        /* prepare the test */
        assert_se(set_unit_path(TEST_DATA_DIR("")) >= 0);
```
I got many `/tmp/fake-xdg-runtime-*`-directories :-)

Maybe, we should call `TEST_DATA_DIR` before `setup_fake_runtime_dir`. As another option we can propagate error from `get_exe_relative_testdata_dir`